tests/unit/customizations/configure/test_exportcreds.py

```diff
@@ -0,0 +1,296 @@
+            "resolution failed")
+        rc = self.export_creds_cmd(args=[], parsed_globals=self.global_args)
+        self.assertIn(
+            'resolution failed', self.err_stream.getvalue()
+        )
+        self.assertEqual(rc, 1)
+
+    def test_can_detect_recursive_resolution(self):
+        self.os_env['_AWS_CLI_PROFILE_CHAIN'] = 'default'
+        rc = self.export_creds_cmd(args=[], parsed_globals=self.global_args)
+        self.assertIn(
+            'Recursive credential resolution process detected',
+            self.err_stream.getvalue()
+        )
+        self.assertEqual(rc, 2)
+
+    def test_nested_calls_not_recursive(self):
+        self.session.get_credentials.return_value = self.creds
+        # If we've seen the profiles foo and bar:
+        self.os_env['_AWS_CLI_PROFILE_CHAIN'] = 'foo,bar'
+        # And we're now on baz, then we shouldn't get an error
+        # because there's no cycle.
+        self.session.get_config_variable.return_value = 'baz'
+        rc = self.export_creds_cmd(args=[], parsed_globals=self.global_args)
+        self.assertEqual(rc, 0)
+        self.assertEqual(
+            self.out_stream.getvalue(),
+            JSONValue(
+                '{"Version": 1, "AccessKeyId": "access_key", '
+                '"SecretAccessKey": "secret_key"}')
+        )
+
+    def test_nested_calls_with_cycle(self):
+        self.session.get_credentials.return_value = self.creds
+        self.os_env['_AWS_CLI_PROFILE_CHAIN'] = 'foo,bar,baz'
+        self.session.get_config_variable.return_value = 'bar'
+        rc = self.export_creds_cmd(args=[], parsed_globals=self.global_args)
+        self.assertIn(
+            'Recursive credential resolution process detected',
+            self.err_stream.getvalue()
+        )
+        self.assertEqual(rc, 2)
+
+    def test_handles_comma_char_in_profile_name_no_cycle(self):
+        self.session.get_credentials.return_value = self.creds
+        self.os_env['_AWS_CLI_PROFILE_CHAIN'] = 'foo,bar,baz'
+        # Shouldn't be a cycle, the comma is part of the profile name.
+        self.session.get_config_variable.return_value = 'bar,baz'
+        rc = self.export_creds_cmd(args=[], parsed_globals=self.global_args)
+        self.assertEqual(rc, 0)
+        self.assertEqual(
+            self.out_stream.getvalue(),
+            JSONValue(
+                '{"Version": 1, "AccessKeyId": "access_key", '
+                '"SecretAccessKey": "secret_key"}')
+        )
+
+    def test_detects_comma_char_with_cycle(self):
+        self.session.get_credentials.return_value = self.creds
+        # To test the round tripping of CSV writing/reading, we'll invoke
+        # the command twice to ensure however it records the profiles
+        # that it detects a cycle with names that need to be escaped/quoted.
+        self.os_env['_AWS_CLI_PROFILE_CHAIN'] = 'foo,bar'
+        self.session.get_config_variable.return_value = 'bar,baz'
```

Comment:
I ended up having separate error messages for each case so I updated the test to check this this is because of a cycle and not max recursion.  The cycle error message now looks like this:

```
$ aws configure export-credentials --profile cycle-a

Unable to retrieve credentials: Error when retrieving credentials from custom-process:
Unable to retrieve credentials: Error when retrieving credentials from custom-process:
Unable to retrieve credentials: Error when retrieving credentials from custom-process:
Credential process resolution detected an infinite loop, profile cycle: cycle-a -> cycle-b -> cycle-c -> cycle-a

```